### ISSUE TITLE
Annotation Processor doesn't handle class generics

### ISSUE THREAD

**Issue by gabizou** [SpongePowered/Mixin#66]
Provided the following:
```java
@Mixin(value = PropertyHelper.class)
public abstract class MixinPropertyHelper<T extends Comparable<T>> implements BlockTrait<T>  {
}
```
and by extension:
```java
@Mixin(value = PropertyBool.class)
public abstract class MixinPropertyBoolean extends MixinPropertyHelper<Boolean> implements BooleanTrait {
}
```
The AP will fail to find the superclass.

**Comment by bloodmc** [comment]
@Mumfrey The issue I am receiving when attempting to build is the following 

https://gist.github.com/bloodmc/c00cef4a8bdbf7e170db

Thanks

**Comment by Mumfrey** [comment]
No worries, gabi mentioned this to me the other day. If you use the AP flag to disable the check for now that'll at least let you build, I'll take a look at this early next week as it should be pretty simple to fix, I just need to strip the generic part off the class name when searching for superclasses.

The issue basically boils down to the the mirror API being utterly insufficient for what we need to do for the most part, there are a lot of things in the specific *implementations* of mirror that I kind of need but don't have direct access to. One of the ways I work around this is by parsing the results of some api object's `toString` methods which emit - say - a class name I need or something like that, and then doing my own processing on the results. This of course causes issues when something I haven't forseen gets emitted by a `toString`, or when different implementations of mirror actually *behave* differently because a specific behaviour isn't stipulated by the API.

This should be pretty straightforward to fix, unfortunately this kind of thing will occasionally crop up because the only alternative would be to have bindings for all possible implementations of the mirror API (which is basically impossible) and get the information we need that way, which is basically not a realistic proposition.

Use the work-around for now, will fix this early next week.

**Comment by Mumfrey** [comment]
Fixed by 34dab7d1d57d02fcd1bf59f0d2235ea035e1b4b2
